pebblo/app/service/local_ui_service.py

```diff
@@ -399,67 +440,64 @@ def get_vector_dbs(chains: dict) -> list:
         return list(set(vector_dbs))
 
     @staticmethod
-    def get_all_documents(retrieval_data: dict) -> dict:
+    def sort_based_on_retrievals(retrieval_data: dict, search_key: str) -> dict:
         """
-        This function returns documents per app in sorted descending order
+        This function returns data based on search key per app in sorted descending order
         based on number of times it appeared in retrievals.
         """
-        documents: dict = {}
-        all_sorted_documents: dict = {}
+        resp: dict = {}
+        sorted_resp: dict = {}
 
-        # fetch document wise retrievals
+        # fetch data wise retrievals
         for data in retrieval_data:
             data_context = data.get("context")
             for context in data_context:
-                document_name = context["retrieved_from"]
-                if document_name in documents.keys():
-                    documents[document_name].extend([data])
+                data_name = context[search_key]
+                if data_name in resp.keys():
+                    resp[data_name].extend([data])
                 else:
-                    documents[document_name] = [data]
+                    resp[data_name] = [data]
 
-        # sorting based on length on retrieval values per documents
-        all_documents = sorted(
-            documents.items(), key=lambda kv: (len(kv[1]), kv[0]), reverse=True
+        # sorting based on length on retrieval values per search key
+        all_resp_data = sorted(
+            resp.items(), key=lambda kv: (len(kv[1]), kv[0]), reverse=True
         )
 
         # converting sorted tuples to dictionary
-        for user_name, data in all_documents:
-            if user_name in all_sorted_documents.keys():
-                all_sorted_documents[user_name].extend(data)
+        for user_name, data in all_resp_data:
```

Comment:
It won't be user_name, it could be any search key present within context object.